CREATE TABLE tripal_gff_temp ( feature_id integer NOT NULL, organism_id integer NOT NULL, uniquename text NOT NULL, type_name character varying(1024) NOT NULL);
CREATE TABLE tripal_gffcds_temp ( feature_id integer NOT NULL, parent_id integer NOT NULL, phase integer, strand integer NOT NULL, fmin integer NOT NULL, fmax integer NOT NULL);
CREATE TABLE tripal_gffprotein_temp ( feature_id integer NOT NULL, parent_id integer NOT NULL, fmin integer NOT NULL, fmax integer NOT NULL);
CREATE TABLE tripal_obo_temp ( id character varying(255) NOT NULL, stanza text NOT NULL, type character varying(50) NOT NULL);
ALTER TABLE ONLY tripal_gff_temp ADD CONSTRAINT tripal_gff_temp__tripal_gff_temp_uq0__key UNIQUE (feature_id);
ALTER TABLE ONLY tripal_gff_temp ADD CONSTRAINT tripal_gff_temp__tripal_gff_temp_uq1__key UNIQUE (uniquename, organism_id, type_name);
ALTER TABLE ONLY tripal_gffprotein_temp ADD CONSTRAINT tripal_gffprotein_temp__tripal_gff_temp_uq0__key UNIQUE (feature_id);
ALTER TABLE ONLY tripal_obo_temp ADD CONSTRAINT tripal_obo_temp__tripal_obo_temp0__key UNIQUE (id);
CREATE INDEX tripal_gff_temp__tripal_gff_temp_idx0__idx ON tripal_gff_temp USING btree (organism_id);
CREATE INDEX tripal_gff_temp__tripal_gff_temp_idx1__idx ON tripal_gff_temp USING btree (uniquename);
CREATE INDEX tripal_gffcds_temp__tripal_gff_temp_idx0__idx ON tripal_gffcds_temp USING btree (parent_id);
CREATE INDEX tripal_gffprotein_temp__tripal_gff_temp_idx0__idx ON tripal_gffprotein_temp USING btree (parent_id);
CREATE INDEX tripal_obo_temp__tripal_obo_temp_idx0__idx ON tripal_obo_temp USING btree (id);
CREATE INDEX tripal_obo_temp__tripal_obo_temp_idx1__idx ON tripal_obo_temp USING btree (type);
CREATE TABLE organism_stock_count ( organism_id bigint NOT NULL, genus character varying(255) NOT NULL, species character varying(255) NOT NULL, common_name character varying(255), num_stocks integer NOT NULL, cvterm_id bigint NOT NULL, stock_type character varying(255) NOT NULL);
CREATE INDEX organism_stock_count__organism_stock_count_idx1__idx ON organism_stock_count USING btree (organism_id);
CREATE INDEX organism_stock_count__organism_stock_count_idx2__idx ON organism_stock_count USING btree (cvterm_id);
CREATE INDEX organism_stock_count__organism_stock_count_idx3__idx ON organism_stock_count USING btree (stock_type);
CREATE TABLE library_feature_count ( library_id bigint NOT NULL, name character varying(255) NOT NULL, num_features integer NOT NULL, feature_type character varying(255) NOT NULL);
CREATE INDEX library_feature_count__library_feature_count_idx1__idx ON library_feature_count USING btree (library_id);
CREATE TABLE organism_feature_count ( organism_id bigint NOT NULL, genus character varying(255) NOT NULL, species character varying(255) NOT NULL, common_name character varying(255), num_features integer NOT NULL, cvterm_id bigint NOT NULL, feature_type character varying(255) NOT NULL);
CREATE INDEX organism_feature_count__organism_feature_count_idx1__idx ON organism_feature_count USING btree (organism_id);
CREATE INDEX organism_feature_count__organism_feature_count_idx2__idx ON organism_feature_count USING btree (cvterm_id);
CREATE INDEX organism_feature_count__organism_feature_count_idx3__idx ON organism_feature_count USING btree (feature_type);
CREATE TABLE analysis_organism ( analysis_id bigint NOT NULL, organism_id bigint NOT NULL);
CREATE INDEX analysis_organism__networkmod_qtl_indx0__idx ON analysis_organism USING btree (analysis_id);
CREATE INDEX analysis_organism__networkmod_qtl_indx1__idx ON analysis_organism USING btree (organism_id);
CREATE TABLE db2cv_mview ( cv_id integer NOT NULL, cvname character varying(255) NOT NULL, db_id integer NOT NULL, dbname character varying(255) NOT NULL, num_terms integer NOT NULL);
CREATE INDEX db2cv_mview__cv_id_idx__idx ON db2cv_mview USING btree (cv_id);
CREATE INDEX db2cv_mview__cvname_idx__idx ON db2cv_mview USING btree (cvname);
CREATE INDEX db2cv_mview__db_id_idx__idx ON db2cv_mview USING btree (db_id);
CREATE INDEX db2cv_mview__dbname_idx__idx ON db2cv_mview USING btree (db_id);
CREATE TABLE cv_root_mview ( name character varying(255) NOT NULL, cvterm_id bigint NOT NULL, cv_id bigint NOT NULL, cv_name character varying(255) NOT NULL);
CREATE INDEX cv_root_mview__cv_root_mview_indx1__idx ON cv_root_mview USING btree (cvterm_id);
CREATE INDEX cv_root_mview__cv_root_mview_indx2__idx ON cv_root_mview USING btree (cv_id);
INSERT INTO db VALUES (2, 'CO_010', 'Crop Germplasm Ontology', 'http://www.cropontology.org/terms/CO_010:{accession}', 'http://www.cropontology.org/get-ontology/CO_010');
INSERT INTO db VALUES (11, 'OBI', 'The Ontology for Biomedical Investigation', 'http://purl.obolibrary.org/obo/{db}_{accession}', 'http://obi-ontology.org/page/Main_Page');
INSERT INTO db VALUES (3, 'dc', 'DCMI Metadata Terms', 'http://purl.org/dc/terms/{accession}', 'http://purl.org/dc/dcmitype/');
INSERT INTO db VALUES (17, 'PMID', 'PubMed', 'http://www.ncbi.nlm.nih.gov/pubmed/{accession}', 'http://www.ncbi.nlm.nih.gov/pubmed');
INSERT INTO db VALUES (12, 'OGI', 'Ontology for genetic interval', 'http://purl.obolibrary.org/obo/{db}_{accession}', 'http://purl.bioontology.org/ontology/OGI');
INSERT INTO db VALUES (33, 'SIO', 'Semanticscience Integrated Ontology', 'http://semanticscience.org/resource/{db}_{accession}', 'http://sio.semanticscience.org/');
INSERT INTO db VALUES (26, 'TCONTACT', 'Tripal Contact Ontology. A temporary ontology until a more formal appropriate ontology an be identified.', 'cv/lookup/TCONTACT/{accession}  ', 'cv/lookup/TCONTACT');
INSERT INTO db VALUES (13, 'IAO', 'Information Artifact Ontology', 'http://purl.obolibrary.org/obo/{db}_{accession}', 'https://github.com/information-artifact-ontology/IAO/');
INSERT INTO db VALUES (4, 'data', 'Bioinformatics operations, data types, formats, identifiers and topics.', 'http://edamontology.org/{db}_{accession}', 'http://edamontology.org/page');
INSERT INTO db VALUES (18, 'UO', 'Units of Measurement Ontology', 'http://purl.obolibrary.org/obo/UO_{accession}', 'http://purl.obolibrary.org/obo/uo');
INSERT INTO db VALUES (5, 'format', 'A defined way or layout of representing and structuring data in a computer file, blob, string, message, or elsewhere. The main focus in EDAM lies on formats as means of structuring data exchanged between different tools or resources.', 'http://edamontology.org/{db}_{accession}', 'http://edamontology.org/page');
INSERT INTO db VALUES (6, 'operation', 'A function that processes a set of inputs and results in a set of outputs, or associates arguments (inputs) with values (outputs). Special cases are: a) An operation that consumes no input (has no input arguments).', 'http://edamontology.org/{db}_{accession}', 'http://edamontology.org/page');
INSERT INTO db VALUES (23, 'GO', 'The Gene Ontology (GO) knowledgebase is the world’s largest source of information on the functions of genes', 'http://amigo.geneontology.org/amigo/term/{db}:{accession}', 'http://geneontology.org/');
INSERT INTO db VALUES (7, 'topic', 'A category denoting a rather broad domain or field of interest, of study, application, work, data, or technology. Topics have no clearly defined borders between each other.', 'http://edamontology.org/{db}_{accession}', 'http://edamontology.org/page');
INSERT INTO db VALUES (8, 'EFO', 'Experimental Factor Ontology', 'http://www.ebi.ac.uk/efo/{db}_{accession}', 'http://www.ebi.ac.uk/efo/efo.owl');
INSERT INTO db VALUES (19, 'NCIT', 'The NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', 'http://purl.obolibrary.org/obo/{db}_{accession}', 'http://purl.obolibrary.org/obo/ncit.owl');
INSERT INTO db VALUES (9, 'ERO', 'The Eagle-I Research Resource Ontology', 'http://purl.bioontology.org/ontology/ERO/{db}:{accession}', 'http://purl.bioontology.org/ontology/ERO');
INSERT INTO db VALUES (30, 'rdf', 'Resource Description Framework', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#', 'http://www.w3.org/1999/02/22-rdf-syntax-ns');
INSERT INTO db VALUES (24, 'SO', 'The Sequence Ontology', 'http://www.sequenceontology.org/browser/current_svn/term/{db}:{accession}', 'http://www.sequenceontology.org');
INSERT INTO db VALUES (10, 'OBCS', 'Ontology of Biological and Clinical Statistics', 'http://purl.obolibrary.org/obo/{db}_{accession}', 'https://github.com/obcs/obcs');
INSERT INTO db VALUES (20, 'NCBITaxon', 'NCBI organismal classification', 'https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id={accession}', 'http://www.berkeleybop.org/ontologies/ncbitaxon/');
INSERT INTO db VALUES (27, 'TPUB', 'Tripal Publication Ontology. A temporary ontology until a more formal appropriate ontology an be identified.', 'cv/lookup/TPUB/{accession}  ', 'cv/lookup/TPUB');
INSERT INTO db VALUES (21, 'rdfs', 'Resource Description Framework Schema', 'http://www.w3.org/2000/01/rdf-schema#{accession}', 'https://www.w3.org/TR/rdf-schema/');
INSERT INTO db VALUES (25, 'TAXRANK', 'A vocabulary of taxonomic ranks (species, family, phylum, etc)', 'http://purl.obolibrary.org/obo/{db}_{accession}', 'http://www.obofoundry.org/ontology/taxrank.html');
INSERT INTO db VALUES (14, 'local', 'Terms created for this site', 'cv/lookup/{db}/{accession}', 'cv/lookup/local');
INSERT INTO db VALUES (22, 'RO', 'Relationship Ontology (legacy)', 'cv/lookup/RO/{accession}    ', 'cv/lookup/RO');
INSERT INTO db VALUES (15, 'SBO', 'Systems Biology Ontology', 'http://purl.obolibrary.org/obo/{db}_{accession}', 'http://www.ebi.ac.uk/sbo/main/');
INSERT INTO db VALUES (28, 'foaf', 'Friend of a Friend', 'http://xmlns.com/foaf/spec/#', 'http://www.foaf-project.org/');
INSERT INTO db VALUES (31, 'schema', 'Schema.org', 'https://schema.org/{accession}', 'https://schema.org/');
INSERT INTO db VALUES (16, 'SWO', 'Bioinformatics operations, data types, formats, identifiers and topics', 'http://www.ebi.ac.uk/swo/{db}_{accession}', 'http://purl.obolibrary.org/obo/swo');
INSERT INTO db VALUES (29, 'hydra', 'A Vocabulary for Hypermedia-Driven Web APIs', 'http://www.w3.org/ns/hydra/core#{accession}', 'http://www.w3.org/ns/hydra/core');
INSERT INTO db VALUES (32, 'sep', 'Sample processing and separation techniques.', 'http://purl.obolibrary.org/obo/{db}_{accession}', 'http://psidev.info/index.php?q=node/312');
SELECT pg_catalog.setval('db_db_id_seq', 33, true);
INSERT INTO dbxref VALUES (3, 2, '0000044', '', NULL);
INSERT INTO dbxref VALUES (4, 2, '0000255', '', NULL);
INSERT INTO dbxref VALUES (5, 2, '0000029', '', NULL);
INSERT INTO dbxref VALUES (6, 2, '0000162', '', NULL);
INSERT INTO dbxref VALUES (7, 3, 'Service', '', NULL);
INSERT INTO dbxref VALUES (8, 4, '1249', '', NULL);
INSERT INTO dbxref VALUES (9, 4, '2190', '', NULL);
INSERT INTO dbxref VALUES (10, 4, '2091', '', NULL);
INSERT INTO dbxref VALUES (11, 4, '2044', '', NULL);
INSERT INTO dbxref VALUES (12, 4, '0849', '', NULL);
INSERT INTO dbxref VALUES (13, 4, '0842', '', NULL);
INSERT INTO dbxref VALUES (14, 4, '2976', '', NULL);
INSERT INTO dbxref VALUES (15, 4, '2968', '', NULL);
INSERT INTO dbxref VALUES (16, 4, '1274', '', NULL);
INSERT INTO dbxref VALUES (17, 4, '1278', '', NULL);
INSERT INTO dbxref VALUES (18, 4, '1280', '', NULL);
INSERT INTO dbxref VALUES (19, 4, '2012', '', NULL);
INSERT INTO dbxref VALUES (20, 4, '1056', '', NULL);
INSERT INTO dbxref VALUES (21, 4, '1048', '', NULL);
INSERT INTO dbxref VALUES (22, 4, '1047', '', NULL);
INSERT INTO dbxref VALUES (23, 4, '2336', '', NULL);
INSERT INTO dbxref VALUES (24, 4, '0853', '', NULL);
INSERT INTO dbxref VALUES (25, 4, '3002', '', NULL);
INSERT INTO dbxref VALUES (26, 4, '0872', '', NULL);
INSERT INTO dbxref VALUES (27, 4, '3272', '', NULL);
INSERT INTO dbxref VALUES (28, 4, '3271', '', NULL);
INSERT INTO dbxref VALUES (29, 6, '0567', '', NULL);
INSERT INTO dbxref VALUES (30, 6, '0564', '', NULL);
INSERT INTO dbxref VALUES (31, 6, '0525', '', NULL);
INSERT INTO dbxref VALUES (32, 6, '0362', '', NULL);
INSERT INTO dbxref VALUES (33, 6, '2945', '', NULL);
INSERT INTO dbxref VALUES (34, 8, '0000548', '', NULL);
INSERT INTO dbxref VALUES (35, 8, '0000269', '', NULL);
INSERT INTO dbxref VALUES (36, 8, '0005522', '', NULL);
INSERT INTO dbxref VALUES (37, 8, '0001728', '', NULL);
INSERT INTO dbxref VALUES (38, 9, '0001716', '', NULL);
INSERT INTO dbxref VALUES (39, 9, '0000387', '', NULL);
INSERT INTO dbxref VALUES (40, 10, '0000117', '', NULL);
INSERT INTO dbxref VALUES (41, 11, '0100026', '', NULL);
INSERT INTO dbxref VALUES (42, 11, '0000070', '', NULL);
INSERT INTO dbxref VALUES (43, 12, '0000021', '', NULL);
INSERT INTO dbxref VALUES (44, 13, '0000115', '', NULL);
INSERT INTO dbxref VALUES (45, 13, '0000129', '', NULL);
INSERT INTO dbxref VALUES (46, 13, '0000064', '', NULL);
INSERT INTO dbxref VALUES (47, 14, 'property', '', NULL);
INSERT INTO dbxref VALUES (48, 14, 'timelastmodified', '', NULL);
INSERT INTO dbxref VALUES (49, 14, 'timeaccessioned', '', NULL);
INSERT INTO dbxref VALUES (50, 14, 'timeexecuted', '', NULL);
INSERT INTO dbxref VALUES (51, 14, 'infraspecific_type', '', NULL);
INSERT INTO dbxref VALUES (52, 14, 'abbreviation', '', NULL);
INSERT INTO dbxref VALUES (53, 14, 'expression', '', NULL);
INSERT INTO dbxref VALUES (54, 14, 'is_analysis', '', NULL);
INSERT INTO dbxref VALUES (55, 14, 'is_obsolete', '', NULL);
INSERT INTO dbxref VALUES (56, 14, 'is_current', '', NULL);
INSERT INTO dbxref VALUES (57, 14, 'is_internal', '', NULL);
INSERT INTO dbxref VALUES (58, 14, 'miniref', '', NULL);
INSERT INTO dbxref VALUES (59, 14, 'array_batch_identifier', '', NULL);
INSERT INTO dbxref VALUES (60, 14, 'relationship_subject', '', NULL);
INSERT INTO dbxref VALUES (61, 14, 'relationship_object', '', NULL);
INSERT INTO dbxref VALUES (62, 14, 'relationship_type', '', NULL);
INSERT INTO dbxref VALUES (63, 14, 'fasta_definition', '', NULL);
INSERT INTO dbxref VALUES (64, 14, 'Reference Feature', '', NULL);
INSERT INTO dbxref VALUES (65, 14, 'fmin', '', NULL);
INSERT INTO dbxref VALUES (66, 14, 'fmax', '', NULL);
INSERT INTO dbxref VALUES (67, 14, 'analysis', '', NULL);
INSERT INTO dbxref VALUES (68, 14, 'source_data', '', NULL);
INSERT INTO dbxref VALUES (69, 14, 'contact', '', NULL);
INSERT INTO dbxref VALUES (70, 14, 'relationship', '', NULL);
INSERT INTO dbxref VALUES (71, 14, 'biomaterial', '', NULL);
INSERT INTO dbxref VALUES (72, 14, 'array_dimensions', '', NULL);
INSERT INTO dbxref VALUES (73, 14, 'element_dimensions', '', NULL);
INSERT INTO dbxref VALUES (74, 14, 'num_of_elements', '', NULL);
INSERT INTO dbxref VALUES (75, 14, 'num_array_columns', '', NULL);
INSERT INTO dbxref VALUES (76, 14, 'num_array_rows', '', NULL);
INSERT INTO dbxref VALUES (77, 14, 'num_grid_columns', '', NULL);
INSERT INTO dbxref VALUES (78, 14, 'num_grid_rows', '', NULL);
INSERT INTO dbxref VALUES (79, 14, 'num_sub_columns', '', NULL);
INSERT INTO dbxref VALUES (80, 14, 'num_sub_rows', '', NULL);
INSERT INTO dbxref VALUES (81, 14, 'Genome Project', '', NULL);
INSERT INTO dbxref VALUES (82, 14, 'rank', '', NULL);
INSERT INTO dbxref VALUES (83, 14, 'lineage', '', NULL);
INSERT INTO dbxref VALUES (84, 14, 'genetic_code_name', '', NULL);
INSERT INTO dbxref VALUES (85, 14, 'mitochondrial_genetic_code', '', NULL);
INSERT INTO dbxref VALUES (86, 14, 'mitochondrial_genetic_code_name', '', NULL);
INSERT INTO dbxref VALUES (87, 14, 'division', '', NULL);
INSERT INTO dbxref VALUES (88, 14, 'genbank_common_name', '', NULL);
INSERT INTO dbxref VALUES (89, 14, 'synonym', '', NULL);
INSERT INTO dbxref VALUES (90, 14, 'other_name', '', NULL);
INSERT INTO dbxref VALUES (91, 14, 'equivalent_name', '', NULL);
INSERT INTO dbxref VALUES (92, 14, 'anamorph', '', NULL);
INSERT INTO dbxref VALUES (93, 14, 'Analysis Type', '', NULL);
INSERT INTO dbxref VALUES (94, 14, 'phylo_leaf', '', NULL);
INSERT INTO dbxref VALUES (95, 14, 'phylo_root', '', NULL);
INSERT INTO dbxref VALUES (96, 14, 'phylo_interior', '', NULL);
INSERT INTO dbxref VALUES (97, 14, 'taxonomy', '', NULL);
INSERT INTO dbxref VALUES (98, 14, 'cM', '', NULL);
INSERT INTO dbxref VALUES (99, 14, 'bp', '', NULL);
INSERT INTO dbxref VALUES (100, 14, 'bin_unit', '', NULL);
INSERT INTO dbxref VALUES (101, 14, 'marker_order', '', NULL);
INSERT INTO dbxref VALUES (102, 14, 'undefined', '', NULL);
INSERT INTO dbxref VALUES (103, 14, 'start', '', NULL);
INSERT INTO dbxref VALUES (104, 14, 'stop', '', NULL);
INSERT INTO dbxref VALUES (105, 14, 'Map Dbxref', '', NULL);
INSERT INTO dbxref VALUES (106, 14, 'Map Type', '', NULL);
INSERT INTO dbxref VALUES (107, 14, 'Genome Group', '', NULL);
INSERT INTO dbxref VALUES (108, 14, 'URL', '', NULL);
INSERT INTO dbxref VALUES (109, 14, 'Population Type', '', NULL);
INSERT INTO dbxref VALUES (110, 14, 'Population Size', '', NULL);
INSERT INTO dbxref VALUES (111, 14, 'Methods', '', NULL);
INSERT INTO dbxref VALUES (112, 14, 'Software', '', NULL);
INSERT INTO dbxref VALUES (113, 14, 'library', '', NULL);
INSERT INTO dbxref VALUES (114, 14, 'library_description', '', NULL);
INSERT INTO dbxref VALUES (115, 14, 'cdna_library', '', NULL);
INSERT INTO dbxref VALUES (116, 14, 'bac_library', '', NULL);
INSERT INTO dbxref VALUES (117, 14, 'fosmid_library', '', NULL);
INSERT INTO dbxref VALUES (118, 14, 'cosmid_library', '', NULL);
INSERT INTO dbxref VALUES (119, 14, 'yac_library', '', NULL);
INSERT INTO dbxref VALUES (120, 14, 'genomic_library', '', NULL);
INSERT INTO dbxref VALUES (121, 14, 'Project Description', '', NULL);
INSERT INTO dbxref VALUES (122, 14, 'Project Type', '', NULL);
INSERT INTO dbxref VALUES (123, 14, 'Study Type', '', NULL);
INSERT INTO dbxref VALUES (124, 14, 'analysis_date', '', NULL);
INSERT INTO dbxref VALUES (125, 14, 'analysis_short_name', '', NULL);
INSERT INTO dbxref VALUES (126, 14, 'Genotyping', '', NULL);
INSERT INTO dbxref VALUES (127, 14, 'Phenotyping', '', NULL);
INSERT INTO dbxref VALUES (128, 14, 'Location', '', NULL);
INSERT INTO dbxref VALUES (129, 15, '0000358', '', NULL);
INSERT INTO dbxref VALUES (130, 15, '0000554', '', NULL);
INSERT INTO dbxref VALUES (131, 15, '0000374', '', NULL);
INSERT INTO dbxref VALUES (132, 16, '0000001', '', NULL);
INSERT INTO dbxref VALUES (133, 18, '0000000', '', NULL);
INSERT INTO dbxref VALUES (134, 19, 'C25164', '', NULL);
INSERT INTO dbxref VALUES (135, 19, 'C48036', '', NULL);
INSERT INTO dbxref VALUES (136, 19, 'C45378', '', NULL);
INSERT INTO dbxref VALUES (137, 19, 'C25712', '', NULL);
INSERT INTO dbxref VALUES (138, 19, 'C44170', '', NULL);
INSERT INTO dbxref VALUES (139, 19, 'C48697', '', NULL);
INSERT INTO dbxref VALUES (140, 19, 'C45559', '', NULL);
INSERT INTO dbxref VALUES (141, 19, 'C80488', '', NULL);
INSERT INTO dbxref VALUES (142, 19, 'C16977', '', NULL);
INSERT INTO dbxref VALUES (143, 19, 'C16631', '', NULL);
INSERT INTO dbxref VALUES (144, 19, 'C25341', '', NULL);
INSERT INTO dbxref VALUES (145, 19, 'C802', '', NULL);
INSERT INTO dbxref VALUES (146, 19, 'C16551', '', NULL);
INSERT INTO dbxref VALUES (147, 19, 'C42765', '', NULL);
INSERT INTO dbxref VALUES (148, 19, 'C15320', '', NULL);
INSERT INTO dbxref VALUES (149, 19, 'C54131', '', NULL);
INSERT INTO dbxref VALUES (150, 19, 'C47885', '', NULL);
INSERT INTO dbxref VALUES (151, 19, 'C16223', '', NULL);
INSERT INTO dbxref VALUES (152, 19, 'C85496', '', NULL);
INSERT INTO dbxref VALUES (153, 19, 'C25693', '', NULL);
INSERT INTO dbxref VALUES (154, 20, 'common_name', '', NULL);
INSERT INTO dbxref VALUES (155, 21, 'comment', '', NULL);
INSERT INTO dbxref VALUES (156, 21, 'type', '', NULL);
INSERT INTO dbxref VALUES (157, 21, 'label', '', NULL);
INSERT INTO dbxref VALUES (158, 24, '0000110', '', NULL);
INSERT INTO dbxref VALUES (159, 24, '0000704', '', NULL);
INSERT INTO dbxref VALUES (160, 24, '0000234', '', NULL);
INSERT INTO dbxref VALUES (161, 24, '0000771', '', NULL);
INSERT INTO dbxref VALUES (162, 24, '0001060', '', NULL);
INSERT INTO dbxref VALUES (163, 24, '0001645', '', NULL);
INSERT INTO dbxref VALUES (164, 24, '0001500', '', NULL);
INSERT INTO dbxref VALUES (165, 25, '0000005', '', NULL);
INSERT INTO dbxref VALUES (166, 25, '0000006', '', NULL);
INSERT INTO dbxref VALUES (167, 25, '0000045', '', NULL);
INSERT INTO dbxref VALUES (168, 27, '0000002', '', NULL);
INSERT INTO dbxref VALUES (169, 29, 'Collection', '', NULL);
INSERT INTO dbxref VALUES (170, 29, 'member', '', NULL);
INSERT INTO dbxref VALUES (171, 29, 'description', '', NULL);
INSERT INTO dbxref VALUES (172, 29, 'totalItems', '', NULL);
INSERT INTO dbxref VALUES (173, 29, 'title', '', NULL);
INSERT INTO dbxref VALUES (174, 29, 'PartialCollectionView', '', NULL);
INSERT INTO dbxref VALUES (175, 31, 'name', '', NULL);
INSERT INTO dbxref VALUES (176, 31, 'alternateName', '', NULL);
INSERT INTO dbxref VALUES (177, 31, 'comment', '', NULL);
INSERT INTO dbxref VALUES (178, 31, 'description', '', NULL);
INSERT INTO dbxref VALUES (179, 31, 'publication', '', NULL);
INSERT INTO dbxref VALUES (180, 31, 'url', '', NULL);
INSERT INTO dbxref VALUES (181, 31, 'additionalType', '', NULL);
INSERT INTO dbxref VALUES (182, 31, 'ItemPage', '', NULL);
INSERT INTO dbxref VALUES (183, 32, '00195', '', NULL);
INSERT INTO dbxref VALUES (184, 32, '00101', '', NULL);
INSERT INTO dbxref VALUES (185, 33, '000493', '', NULL);
INSERT INTO dbxref VALUES (186, 33, '000631', '', NULL);
INSERT INTO dbxref VALUES (187, 33, '000056', '', NULL);
INSERT INTO dbxref VALUES (188, 33, '001166', '', NULL);
INSERT INTO dbxref VALUES (189, 33, '000281', '', NULL);
INSERT INTO dbxref VALUES (190, 33, '001080', '', NULL);
INSERT INTO dbxref VALUES (191, 33, '001323', '', NULL);
INSERT INTO dbxref VALUES (192, 33, '001007', '', NULL);
INSERT INTO dbxref VALUES (193, 33, '010054', '', NULL);
INSERT INTO dbxref VALUES (194, 33, '001066', '', NULL);
SELECT pg_catalog.setval('dbxref_dbxref_id_seq', 194, true);
INSERT INTO cv VALUES (3, 'Statistical Terms', 'Locally created terms for statistics');
INSERT INTO cv VALUES (5, 'germplasm_ontology', 'GCP germplasm ontology');
INSERT INTO cv VALUES (6, 'dc', 'DCMI Metadata Terms');
INSERT INTO cv VALUES (7, 'EDAM', 'Bioscientific data analysis ontology');
INSERT INTO cv VALUES (8, 'efo', 'The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org');
INSERT INTO cv VALUES (9, 'ero', 'The Eagle-I Research Resource Ontology models research resources such instruments. protocols, reagents, animal models and biospecimens. It has been developed in the context of the eagle-i project (http://eagle-i.net/).');
INSERT INTO cv VALUES (10, 'OBCS', 'Ontology of Biological and Clinical Statistics');
INSERT INTO cv VALUES (11, 'obi', 'Ontology for Biomedical Investigation. The Ontology for Biomedical Investigations (OBI) is build in a collaborative, international effort and will serve as a resource for annotating biomedical investigations, including the study design, protocols and instrumentation used, the data generated and the types of analysis performed on the data. This ontology arose from the Functional Genomics Investigation Ontology (FuGO) and will contain both terms that are common to all biomedical investigations, including functional genomics investigations and those that are more domain specific');
INSERT INTO cv VALUES (12, 'ogi', 'Ontology for Biomedical Investigation. The Ontology for Biomedical Investigations (OBI) is build in a collaborative, international effort and will serve as a resource for annotating biomedical investigations, including the study design, protocols and instrumentation used, the data generated and the types of analysis performed on the data. This ontology arose from the Functional Genomics Investigation Ontology (FuGO) and will contain both terms that are common to all biomedical investigations, including functional genomics investigations and those that are more domain specific');
INSERT INTO cv VALUES (13, 'IAO', 'Information Artifact Ontology');
INSERT INTO cv VALUES (2, 'local', 'Locally created terms');
INSERT INTO cv VALUES (14, 'organism_property', 'A local vocabulary that contains locally defined properties for organisms');
INSERT INTO cv VALUES (15, 'analysis_property', 'A local vocabulary that contains locally defined properties for analyses');
INSERT INTO cv VALUES (16, 'tripal_phylogeny', 'Terms used by the Tripal phylotree module for phylogenetic and taxonomic trees');
INSERT INTO cv VALUES (17, 'feature_relationship', 'A local vocabulary that contains types of relationships between features');
INSERT INTO cv VALUES (18, 'feature_property', 'A local vocabulary that contains properties for genomic features');
INSERT INTO cv VALUES (19, 'contact_property', 'A local vocabulary that contains properties for contacts. This can be used if the tripal_contact vocabulary (which is default for contacts in Tripal) is not desired.');
INSERT INTO cv VALUES (20, 'contact_type', 'A local vocabulary that contains types of contacts. This can be used if the tripal_contact vocabulary (which is default for contacts in Tripal) is not desired.');
INSERT INTO cv VALUES (31, 'tripal_pub', 'Tripal Publication Ontology');
INSERT INTO cv VALUES (22, 'contact_relationship', 'A local vocabulary that contains types of relationships between contacts.');
INSERT INTO cv VALUES (23, 'featuremap_units', 'A local vocabulary that contains map unit types for the unittype_id column of the featuremap table.');
INSERT INTO cv VALUES (24, 'featurepos_property', 'A local vocabulary that contains terms map properties.');
INSERT INTO cv VALUES (25, 'featuremap_property', 'A local vocabulary that contains positional types for the feature positions.');
INSERT INTO cv VALUES (26, 'library_property', 'A local vocabulary that contains properties for libraries.');
INSERT INTO cv VALUES (27, 'library_type', 'A local vocabulary that contains terms for types of libraries (e.g. BAC, cDNA, FOSMID, etc).');
INSERT INTO cv VALUES (28, 'project_property', 'A local vocabulary that contains properties for projects.');
INSERT INTO cv VALUES (29, 'study_property', 'A local vocabulary that contains properties for studies.');
INSERT INTO cv VALUES (30, 'project_relationship', 'A local vocabulary that contains Types of relationships between projects');
INSERT INTO cv VALUES (32, 'pub_type', 'A local vocabulary that contains types of publications. This can be used if the tripal_pub vocabulary (which is default for publications in Tripal) is not desired.');
INSERT INTO cv VALUES (33, 'pub_property', 'A local vocabulary that contains properties for publications. This can be used if the tripal_pub vocabulary (which is default for publications in Tripal) is not desired.');
INSERT INTO cv VALUES (34, 'pub_relationship', 'A local vocabulary that contains types of relationships between publications.');
INSERT INTO cv VALUES (35, 'stock_relationship', 'A local vocabulary that contains types of relationships between stocks.');
INSERT INTO cv VALUES (36, 'stock_property', 'A local vocabulary that contains properties for stocks.');
INSERT INTO cv VALUES (37, 'stock_type', 'A local vocabulary that contains a list of types for stocks.');
INSERT INTO cv VALUES (38, 'tripal_analysis', 'A local vocabulary that contains terms used for analyses.');
INSERT INTO cv VALUES (39, 'nd_experiment_types', 'A local vocabulary that contains terms used for the Natural Diverisity module''s experiment types.');
INSERT INTO cv VALUES (40, 'nd_geolocation_property', 'A local vocabulary that contains terms used for the Natural Diverisity module''s geolocation property.');
INSERT INTO cv VALUES (41, 'sbo', 'Systems Biology.  Terms commonly used in Systems Biology, and in particular in computational modeling.');
INSERT INTO cv VALUES (42, 'swo', 'Bioinformatics operations, data types, formats, identifiers and topics');
INSERT INTO cv VALUES (43, 'PMID', 'PubMed.');
INSERT INTO cv VALUES (44, 'uo', 'Units of Measurement Ontology');
INSERT INTO cv VALUES (45, 'ncit', 'NCI Thesaurus OBO Edition');
INSERT INTO cv VALUES (46, 'ncbitaxon', 'NCBI organismal classification. An ontology representation of the NCBI organismal taxonomy');
INSERT INTO cv VALUES (47, 'rdfs', 'Resource Description Framework Schema');
INSERT INTO cv VALUES (48, 'ro', 'Relationship Ontology (legacy)');
INSERT INTO cv VALUES (49, 'cellular_component', 'Gene Ontology Cellular Component Vocabulary');
INSERT INTO cv VALUES (50, 'biological_process', 'Gene Ontology Biological Process Vocabulary');
INSERT INTO cv VALUES (21, 'tripal_contact', 'Tripal Contact Ontology');
INSERT INTO cv VALUES (51, 'molecular_function', 'Gene Ontology Molecular Function Vocabulary');
INSERT INTO cv VALUES (52, 'sequence', 'The Sequence Ontology');
INSERT INTO cv VALUES (53, 'taxonomic_rank', 'Taxonomic Rank');
INSERT INTO cv VALUES (54, 'foaf', 'Friend of a Friend. A dictionary of people-related terms that can be used in structured data).');
INSERT INTO cv VALUES (55, 'hydra', 'A Vocabulary for Hypermedia-Driven Web APIs.');
INSERT INTO cv VALUES (56, 'rdf', 'Resource Description Framework');
INSERT INTO cv VALUES (57, 'schema', 'Schema.org. Schema.org is sponsored by Google, Microsoft, Yahoo and Yandex. The vocabularies are developed by an open community process.');
INSERT INTO cv VALUES (58, 'sep', 'A structured controlled vocabulary for the annotation of sample processing and separation techniques in scientific experiments.');
INSERT INTO cv VALUES (59, 'SIO', 'The Semanticscience Integrated Ontology (SIO) provides a simple, integrated ontology of types and relations for rich description of objects, processes and their attributes.');
SELECT pg_catalog.setval('cv_cv_id_seq', 59, true);
INSERT INTO cvterm VALUES (3, 5, 'accession', '', 3, 0, 0);
INSERT INTO cvterm VALUES (4, 5, 'generated germplasm', '', 4, 0, 0);
INSERT INTO cvterm VALUES (5, 5, 'cultivar', '', 5, 0, 0);
INSERT INTO cvterm VALUES (6, 5, '414 inbred line', '', 6, 0, 0);
INSERT INTO cvterm VALUES (7, 6, 'Service', 'A system that provides one or more functions.', 7, 0, 0);
INSERT INTO cvterm VALUES (8, 7, 'Sequence length', 'The size (length) of a sequence, subsequence or region in a sequence, or range(s) of lengths.', 8, 0, 0);
INSERT INTO cvterm VALUES (9, 7, 'Sequence checksum', 'A fixed-size datum calculated (by using a hash function) for a molecular sequence, typically for purposes of error detection or indexing.', 9, 0, 0);
INSERT INTO cvterm VALUES (10, 7, 'Accession', 'A persistent (stable) and unique identifier, typically identifying an object (entry) from a database.', 10, 0, 0);
INSERT INTO cvterm VALUES (11, 7, 'Sequence', 'One or more molecular sequences, possibly with associated annotation.', 11, 0, 0);
INSERT INTO cvterm VALUES (12, 7, 'Sequence record', 'A molecular sequence and associated metadata.', 12, 0, 0);
INSERT INTO cvterm VALUES (13, 7, 'Identifier', 'A text token, number or something else which identifies an entity, but which may not be persistent (stable) or unique (the same identifier may identify multiple things).', 13, 0, 0);
INSERT INTO cvterm VALUES (14, 7, 'Protein sequence', 'One or more protein sequences, possibly with associated annotation.', 14, 0, 0);
INSERT INTO cvterm VALUES (15, 7, 'Image', 'Biological or biomedical data has been rendered into an image, typically for display on screen.', 15, 0, 0);
INSERT INTO cvterm VALUES (16, 7, 'Map', 'A map of (typically one) DNA sequence annotated with positional or non-positional features.', 16, 0, 0);
INSERT INTO cvterm VALUES (17, 7, 'Genetic map', 'A map showing the relative positions of genetic markers in a nucleic acid sequence, based on estimation of non-physical distance such as recombination frequencies.', 17, 0, 0);
INSERT INTO cvterm VALUES (18, 7, 'Physical map', 'A map of DNA (linear or circular) annotated with physical features or landmarks such as restriction sites, cloned DNA fragments, genes or genetic markers, along with the physical distances between them. Distance in a physical map is measured in base pairs. A physical map might be ordered relative to a reference map (typically a genetic map) in the process of genome sequencing.', 18, 0, 0);
INSERT INTO cvterm VALUES (19, 7, 'Sequence coordinates', 'A position in a map (for example a genetic map), either a single position (point) or a region / interval.', 19, 0, 0);
INSERT INTO cvterm VALUES (20, 7, 'Database name', 'The name of a biological or bioinformatics database.', 20, 0, 0);
INSERT INTO cvterm VALUES (21, 7, 'Database ID', 'An identifier of a biological or bioinformatics database.', 21, 0, 0);
INSERT INTO cvterm VALUES (22, 7, 'URI', 'The name of a biological or bioinformatics database.', 22, 0, 0);
INSERT INTO cvterm VALUES (23, 7, 'Translation phase specification', 'Phase for translation of DNA (0, 1 or 2) relative to a fragment of the coding sequence.', 23, 0, 0);
INSERT INTO cvterm VALUES (24, 7, 'DNA sense specification', 'The strand of a DNA sequence (forward or reverse).', 24, 0, 0);
INSERT INTO cvterm VALUES (25, 7, 'Annotation track', 'Annotation of one particular positional feature on a biomolecular (typically genome) sequence, suitable for import and display in a genome browser. Synonym: Sequence annotation track.', 25, 0, 0);
INSERT INTO cvterm VALUES (26, 7, 'Phylogenetic tree', 'The raw data (not just an image) from which a phylogenetic tree is directly generated or plotted, such as topology, lengths (in time or in expected amounts of variance) and a confidence interval for each length.', 26, 0, 0);
INSERT INTO cvterm VALUES (27, 7, 'Species tree', 'A phylogenetic tree that reflects phylogeny of the taxa from which the characters (used in calculating the tree) were sampled.', 27, 0, 0);
INSERT INTO cvterm VALUES (28, 7, 'Gene tree', 'A phylogenetic tree that is an estimate of the character''s phylogeny.', 28, 0, 0);
INSERT INTO cvterm VALUES (29, 7, 'Phylogenetic tree visualisation', 'A phylogenetic tree that is an estimate of the character''s phylogeny.', 29, 0, 0);
INSERT INTO cvterm VALUES (30, 7, 'Sequence visualisation', 'Visualise, format or render a molecular sequence or sequences such as a sequence alignment, possibly with sequence features or properties shown.', 30, 0, 0);
INSERT INTO cvterm VALUES (31, 7, 'genome assembly', '', 31, 0, 0);
INSERT INTO cvterm VALUES (32, 7, 'Genome annotation ', '', 32, 0, 0);
INSERT INTO cvterm VALUES (33, 7, 'Analysis', 'Apply analytical methods to existing data of a specific type.', 33, 0, 0);
INSERT INTO cvterm VALUES (34, 8, 'instrument', 'An instrument is a device which provides a mechanical or electronic function.', 34, 0, 0);
INSERT INTO cvterm VALUES (36, 8, 'substrate type', 'Controlled terms for descriptors of types of array substrates.', 36, 0, 0);
INSERT INTO cvterm VALUES (37, 8, 'array manufacturer', '', 37, 0, 0);
INSERT INTO cvterm VALUES (35, 8, 'array design', 'An instrument design which describes the design of the array.', 35, 0, 0);
INSERT INTO cvterm VALUES (38, 9, 'database', 'A database is an organized collection of data, today typically in digital form.', 38, 0, 0);
INSERT INTO cvterm VALUES (39, 9, 'data acquisition', 'A technique that samples real world physical conditions and conversion of the resulting samples into digital numeric values that can be manipulated by a computer.', 39, 0, 0);
INSERT INTO cvterm VALUES (40, 10, 'rank order', 'A data item that represents an arrangement according to a rank, i.e., the position of a particular case relative to other cases on a defined scale.', 40, 0, 0);
INSERT INTO cvterm VALUES (41, 11, 'organism', 'A material entity that is an individual living system, such as animal, plant, bacteria or virus, that is capable of replicating or reproducing, growth and maintenance in the right environment. An organism may be unicellular or made up, like humans, of many billions of cells divided into specialized tissues and organs.', 41, 0, 0);
INSERT INTO cvterm VALUES (42, 11, 'assay', 'A planned process with the objective to produce information about the material entity that is the evaluant, by physically examining it or its proxies.', 42, 0, 0);
INSERT INTO cvterm VALUES (43, 12, 'location on map', '', 43, 0, 0);
INSERT INTO cvterm VALUES (44, 13, 'definition', 'The official OBI definition, explaining the meaning of a class or property. Shall be Aristotelian, formalized and normalized. Can be augmented with colloquial definitions.', 44, 0, 0);
INSERT INTO cvterm VALUES (45, 13, 'version number', 'A version number is an information content entity which is a sequence of characters borne by part of each of a class of manufactured products or its packaging and indicates its order within a set of other products having the same name.', 45, 0, 0);
INSERT INTO cvterm VALUES (46, 13, 'algorithm', 'An algorithm is a set of instructions for performing a paticular calculation.', 46, 0, 0);
INSERT INTO cvterm VALUES (47, 2, 'property', 'A generic term indicating that represents an attribute, quality or characteristic of something.', 47, 0, 0);
INSERT INTO cvterm VALUES (48, 2, 'time_last_modified', 'The time at which the record was last modified.', 48, 0, 0);
INSERT INTO cvterm VALUES (49, 2, 'time_accessioned', 'The time at which the record was first added.', 49, 0, 0);
INSERT INTO cvterm VALUES (50, 2, 'time_executed', 'The time when the task was executed.', 50, 0, 0);
INSERT INTO cvterm VALUES (51, 2, 'infraspecific_type', 'The connector type (e.g. subspecies, varietas, forma, etc.) for the infraspecific name', 51, 0, 0);
INSERT INTO cvterm VALUES (52, 2, 'abbreviation', 'A shortened name (or abbreviation) for the item.', 52, 0, 0);
INSERT INTO cvterm VALUES (53, 2, 'expression', 'Curated expression data', 53, 0, 0);
INSERT INTO cvterm VALUES (54, 2, 'is_analysis', 'Indicates if this feature was predicted computationally using another feature.', 54, 0, 0);
INSERT INTO cvterm VALUES (55, 2, 'is_obsolete', 'Indicates if this record is obsolete.', 55, 0, 0);
INSERT INTO cvterm VALUES (56, 2, 'is_current', 'Indicates if this record is current.', 56, 0, 0);
INSERT INTO cvterm VALUES (57, 2, 'is_internal', 'Indicates if this record is internal and not normally available outside of a local setting.', 57, 0, 0);
INSERT INTO cvterm VALUES (58, 2, 'Mini-ref', 'A small in-house unique identifier for a publication.', 58, 0, 0);
INSERT INTO cvterm VALUES (59, 2, 'Array Batch Identifier', 'A unique identifier for an array batch.', 59, 0, 0);
INSERT INTO cvterm VALUES (60, 2, 'clause subject', 'The subject of a relationship clause.', 60, 0, 0);
INSERT INTO cvterm VALUES (61, 2, 'clause predicate', 'The object of a relationship clause.', 61, 0, 0);
INSERT INTO cvterm VALUES (62, 2, 'relationship type', 'The relationship type.', 62, 0, 0);
INSERT INTO cvterm VALUES (63, 2, 'fasta_definition', 'The definition line for a FASTA formatted sequence', 63, 0, 0);
INSERT INTO cvterm VALUES (64, 2, 'Reference Feature', 'A genomic or genetic feature on which other features are mapped.', 64, 0, 0);
INSERT INTO cvterm VALUES (65, 2, 'minimal boundary', 'The leftmost, minimal boundary in the linear range represented by the feature location. Sometimes this is called start although this is confusing because it does not necessarily represent the 5-prime coordinate.', 65, 0, 0);
INSERT INTO cvterm VALUES (66, 2, 'maximal boundary', 'The rightmost, maximal boundary in the linear range represented by the featureloc. Sometimes this is called end although this is confusing because it does not necessarily represent the 3-prime coordinate', 66, 0, 0);
INSERT INTO cvterm VALUES (67, 2, 'analysis', 'A process as a method of studying the nature of something or of determining its essential features and their relations. (Random House Kernerman Webster''s College Dictionary, © 2010 K Dictionaries Ltd).', 67, 0, 0);
INSERT INTO cvterm VALUES (68, 2, 'source_data', 'The location where data that is being used come from.', 68, 0, 0);
INSERT INTO cvterm VALUES (69, 2, 'contact', 'An entity (e.g. individual or organization) through whom a person can gain access to information, favors, influential people, and the like.', 69, 0, 0);
INSERT INTO cvterm VALUES (70, 2, 'relationship', 'The way in which two things are connected.', 70, 0, 0);
INSERT INTO cvterm VALUES (71, 2, 'biomaterial', 'A biomaterial represents the MAGE concept of BioSource, BioSample, and LabeledExtract. It is essentially some biological material (tissue, cells, serum) that may have been processed. Processed biomaterials should be traceable back to raw biomaterials via the biomaterialrelationship table.', 71, 0, 0);
INSERT INTO cvterm VALUES (72, 2, 'array_dimensions', 'The dimensions of an array.', 72, 0, 0);
INSERT INTO cvterm VALUES (73, 2, 'element_dimensions', 'The dimensions of an element.', 73, 0, 0);
INSERT INTO cvterm VALUES (74, 2, 'num_of_elements', 'The number of elements.', 74, 0, 0);
INSERT INTO cvterm VALUES (75, 2, 'num_array_columns', 'The number of columns in an array.', 75, 0, 0);
INSERT INTO cvterm VALUES (76, 2, 'num_array_rows', 'The number of rows in an array.', 76, 0, 0);
INSERT INTO cvterm VALUES (77, 2, 'num_grid_columns', 'The number of columns in a grid.', 77, 0, 0);
INSERT INTO cvterm VALUES (78, 2, 'num_grid_rows', 'The number of rows in a grid.', 78, 0, 0);
INSERT INTO cvterm VALUES (79, 2, 'num_sub_columns', 'The number of sub columns.', 79, 0, 0);
INSERT INTO cvterm VALUES (80, 2, 'num_sub_rows', 'The number of sub rows.', 80, 0, 0);
INSERT INTO cvterm VALUES (81, 2, 'Genome Project', 'A project for whole genome analysis that can include assembly and annotation.', 81, 0, 0);
INSERT INTO cvterm VALUES (82, 14, 'rank', 'A taxonmic rank', 82, 0, 0);
INSERT INTO cvterm VALUES (83, 14, 'lineage', '', 83, 0, 0);
INSERT INTO cvterm VALUES (84, 14, 'genetic_code_name', '', 84, 0, 0);
INSERT INTO cvterm VALUES (85, 14, 'mitochondrial_genetic_code', '', 85, 0, 0);
INSERT INTO cvterm VALUES (86, 14, 'mitochondrial_genetic_code_name', '', 86, 0, 0);
INSERT INTO cvterm VALUES (87, 14, 'division', '', 87, 0, 0);
INSERT INTO cvterm VALUES (88, 14, 'genbank_common_name', '', 88, 0, 0);
INSERT INTO cvterm VALUES (89, 14, 'synonym', '', 89, 0, 0);
INSERT INTO cvterm VALUES (90, 14, 'other_name', '', 90, 0, 0);
INSERT INTO cvterm VALUES (91, 14, 'equivalent_name', '', 91, 0, 0);
INSERT INTO cvterm VALUES (92, 14, 'anamorph', '', 92, 0, 0);
INSERT INTO cvterm VALUES (93, 15, 'Analysis Type', 'The type of analysis that was performed.', 93, 0, 0);
INSERT INTO cvterm VALUES (94, 16, 'phylo_leaf', 'A leaf node in a phylogenetic tree.', 94, 0, 0);
INSERT INTO cvterm VALUES (95, 16, 'phylo_root', 'The root node of a phylogenetic tree.', 95, 0, 0);
INSERT INTO cvterm VALUES (96, 16, 'phylo_interior', 'An interior node in a phylogenetic tree.', 96, 0, 0);
INSERT INTO cvterm VALUES (97, 16, 'taxonomy', 'Deprecated: A term used to indicate if a phylotree is a taxonomic tree', 97, 0, 0);
INSERT INTO cvterm VALUES (98, 23, 'cM', 'Centimorgan units', 98, 0, 0);
INSERT INTO cvterm VALUES (99, 23, 'bp', 'Base pairs units', 99, 0, 0);
INSERT INTO cvterm VALUES (100, 23, 'bin_unit', 'The bin unit', 100, 0, 0);
INSERT INTO cvterm VALUES (101, 23, 'marker_order', 'Units simply to define marker order.', 101, 0, 0);
INSERT INTO cvterm VALUES (102, 23, 'undefined', 'A catch-all for an undefined unit type', 102, 0, 0);
INSERT INTO cvterm VALUES (103, 24, 'start', 'The start coordinate for a map feature.', 103, 0, 0);
INSERT INTO cvterm VALUES (104, 24, 'stop', 'The end coordinate for a map feature', 104, 0, 0);
INSERT INTO cvterm VALUES (105, 25, 'Map Dbxref', 'A unique identifer for the map in a remote database.  The format is a database abbreviation and a unique accession separated by a colon.  (e.g. Gramene:tsh1996a)', 105, 0, 0);
INSERT INTO cvterm VALUES (106, 25, 'Map Type', 'The type of Map (e.g. QTL, Physical, etc.)', 106, 0, 0);
INSERT INTO cvterm VALUES (107, 25, 'Genome Group', '', 107, 0, 0);
INSERT INTO cvterm VALUES (108, 25, 'URL', 'A univeral resource locator (URL) reference where the publication can be found.  For maps found online, this would be the web address for the map.', 108, 0, 0);
INSERT INTO cvterm VALUES (109, 25, 'Population Type', 'A brief description of the population type used to generate the map (e.g. RIL, F2, BC1, etc).', 109, 0, 0);
INSERT INTO cvterm VALUES (110, 25, 'Population Size', 'The size of the population used to construct the map.', 110, 0, 0);
INSERT INTO cvterm VALUES (111, 25, 'Methods', 'A brief description of the methods used to construct the map.', 111, 0, 0);
INSERT INTO cvterm VALUES (112, 25, 'Software', 'The software used to construct the map.', 112, 0, 0);
INSERT INTO cvterm VALUES (113, 26, 'Library', 'A group of physical entities organized into a collection', 113, 0, 0);
INSERT INTO cvterm VALUES (114, 26, 'Library Description', 'Description of a library', 114, 0, 0);
INSERT INTO cvterm VALUES (115, 27, 'cdna_library', 'cDNA library', 115, 0, 0);
INSERT INTO cvterm VALUES (116, 27, 'bac_library', 'Bacterial Artifical Chromsome (BAC) library', 116, 0, 0);
INSERT INTO cvterm VALUES (117, 27, 'fosmid_library', 'Fosmid library', 117, 0, 0);
INSERT INTO cvterm VALUES (118, 27, 'cosmid_library', 'Cosmid library', 118, 0, 0);
INSERT INTO cvterm VALUES (119, 27, 'yac_library', 'Yeast Artificial Chromosome (YAC) library', 119, 0, 0);
INSERT INTO cvterm VALUES (120, 27, 'genomic_library', 'Genomic Library', 120, 0, 0);
INSERT INTO cvterm VALUES (121, 28, 'Project Description', 'Description of a project', 121, 0, 0);
INSERT INTO cvterm VALUES (122, 28, 'Project Type', 'A type of project', 122, 0, 0);
INSERT INTO cvterm VALUES (123, 29, 'Study Type', 'A type of study', 123, 0, 0);
INSERT INTO cvterm VALUES (124, 38, 'analysis_date', 'The date that an analysis was performed.', 124, 0, 0);
INSERT INTO cvterm VALUES (125, 38, 'analysis_short_name', 'A computer legible (no spaces or special characters) abbreviation for the analysis.', 125, 0, 0);
INSERT INTO cvterm VALUES (126, 39, 'Genotyping', 'An experiment where genotypes of individuals are identified.', 126, 0, 0);
INSERT INTO cvterm VALUES (127, 39, 'Phenotyping', 'An experiment where phenotypes of individuals are identified.', 127, 0, 0);
INSERT INTO cvterm VALUES (128, 40, 'Location', 'The name of the location.', 128, 0, 0);
INSERT INTO cvterm VALUES (129, 41, 'phenotype', 'A biochemical network can generate phenotypes or affects biological processes. Such processes can take place at different levels and are independent of the biochemical network itself.', 129, 0, 0);
INSERT INTO cvterm VALUES (130, 41, 'database cross reference', 'An annotation which directs one to information contained within a database.', 130, 0, 0);
INSERT INTO cvterm VALUES (131, 41, 'relationship', 'Connectedness between entities and/or interactions representing their relatedness or influence.', 131, 0, 0);
INSERT INTO cvterm VALUES (132, 42, 'software', 'Computer software, or generally just software, is any set of machine-readable instructions (most often in the form of a computer program) that conform to a given syntax (sometimes referred to as a language) that is interpretable by a given processor and that directs a computer''s processor to perform specific operations.', 132, 0, 0);
INSERT INTO cvterm VALUES (133, 44, 'unit', 'A unit of measurement is a standardized quantity of a physical quality.', 133, 0, 0);
INSERT INTO cvterm VALUES (134, 45, 'Date', 'The particular day, month and year an event has happened or will happen.', 134, 0, 0);
INSERT INTO cvterm VALUES (135, 45, 'Operator', 'A person that operates some apparatus or machine', 135, 0, 0);
INSERT INTO cvterm VALUES (136, 45, 'Technology Platform', 'The specific version (manufacturer, model, etc.) of a technology that is used to carry out a laboratory or computational experiment.', 136, 0, 0);
INSERT INTO cvterm VALUES (137, 45, 'Value', 'A numerical quantity measured or assigned or computed.', 137, 0, 0);
INSERT INTO cvterm VALUES (138, 45, 'Channel', 'An independent acquisition scheme, i.e., a route or conduit through which flows data consisting of one particular measurement using one particular parameter.', 138, 0, 0);
INSERT INTO cvterm VALUES (139, 45, 'Controlled Vocabulary', 'A set of terms that are selected and defined based on the requirements set out by the user group, usually a set of vocabulary is chosen to promote consistency across data collection projects. [ NCI ]', 139, 0, 0);
INSERT INTO cvterm VALUES (140, 45, 'Term', 'A word or expression used for some particular thing. [ NCI ]', 140, 0, 0);
INSERT INTO cvterm VALUES (141, 45, 'Expression', 'A combination of symbols that represents a value. [ NCI ]', 141, 0, 0);
INSERT INTO cvterm VALUES (142, 45, 'Phenotype', 'The assemblage of traits or outward appearance of an individual. It is the product of interactions between genes and between genes and the environment. [ NCI ]', 142, 0, 0);
INSERT INTO cvterm VALUES (143, 45, 'Genotype', 'The genetic constitution of an organism or cell, as distinct from its expressed features or phenotype. [ NCI ]', 143, 0, 0);
INSERT INTO cvterm VALUES (144, 45, 'Location', 'A position, site, or point in space where something can be found. [ NCI ]', 144, 0, 0);
INSERT INTO cvterm VALUES (145, 45, 'Reagent', 'Any natural or synthetic substance used in a chemical or biological reaction in order to produce, identify, or measure another substance. [ NCI ]', 145, 0, 0);
INSERT INTO cvterm VALUES (146, 45, 'Environment', 'The totality of surrounding conditions. [ NCI ]', 146, 0, 0);
INSERT INTO cvterm VALUES (147, 45, 'Tree Node', 'A term that refers to any individual item or entity in a hierarchy or pedigree. [ NCI ]', 147, 0, 0);
INSERT INTO cvterm VALUES (148, 45, 'Study Design', 'A plan detailing how a study will be performed in order to represent the phenomenon under examination, to answer the research questions that have been asked, and defining the methods of data analysis. Study design is driven by research hypothesis being posed, study subject/population/sample available, logistics/resources: technology, support, networking, collaborative support, etc. [ NCI ]', 148, 0, 0);
INSERT INTO cvterm VALUES (149, 45, 'Company', 'Any formal business entity for profit, which may be a corporation, a partnership, association or individual proprietorship.', 149, 0, 0);
INSERT INTO cvterm VALUES (150, 45, 'Project', 'Any specifically defined piece of work that is undertaken or attempted to meet a single requirement.', 150, 0, 0);
INSERT INTO cvterm VALUES (151, 45, 'DNA Library', 'A collection of DNA molecules that have been cloned in vectors.', 151, 0, 0);
INSERT INTO cvterm VALUES (152, 45, 'Trait', 'Any genetically determined characteristic.', 152, 0, 0);
INSERT INTO cvterm VALUES (153, 45, 'Subgroup', 'A subdivision of a larger group with members often exhibiting similar characteristics. [ NCI ]', 153, 0, 0);
INSERT INTO cvterm VALUES (154, 46, 'common name', '', 154, 0, 0);
INSERT INTO cvterm VALUES (155, 47, 'comment', 'A human-readable description of a resource.', 155, 0, 0);
INSERT INTO cvterm VALUES (156, 47, 'type', 'The type of resource.', 156, 0, 0);
INSERT INTO cvterm VALUES (157, 47, 'label', 'A human-readable version of a resource''s name.', 157, 0, 0);
INSERT INTO cvterm VALUES (158, 52, 'sequence_feature', 'Any extent of continuous biological sequence.', 158, 0, 0);
INSERT INTO cvterm VALUES (159, 52, 'gene', 'A region (or regions) that includes all of the sequence elements necessary to encode a functional transcript. A gene may include regulatory regions, transcribed regions and/or other functional sequence regions.', 159, 0, 0);
INSERT INTO cvterm VALUES (160, 52, 'mRNA', 'Messenger RNA is the intermediate molecule between DNA and protein. It includes UTR and coding sequences. It does not contain introns.', 160, 0, 0);
INSERT INTO cvterm VALUES (161, 52, 'QTL', 'A quantitative trait locus (QTL) is a polymorphic locus which contains alleles that differentially affect the expression of a continuously distributed phenotypic trait. Usually it is a marker described by statistical association to quantitative variation in the particular phenotypic trait that is thought to be controlled by the cumulative action of alleles at multiple loci.', 161, 0, 0);
INSERT INTO cvterm VALUES (162, 52, 'sequence_variant', 'A sequence_variant is a non exact copy of a sequence_feature or genome exhibiting one or more sequence_alteration.', 162, 0, 0);
INSERT INTO cvterm VALUES (163, 52, 'genetic_marker', 'A measurable sequence feature that varies within a population.', 163, 0, 0);
INSERT INTO cvterm VALUES (164, 52, 'heritable_phenotypic_marker', 'A biological_region characterized as a single heritable trait in a phenotype screen. The heritable phenotype may be mapped to a chromosome but generally has not been characterized to a specific gene locus.', 164, 0, 0);
INSERT INTO cvterm VALUES (165, 53, 'genus', '', 165, 0, 0);
INSERT INTO cvterm VALUES (166, 53, 'species', '', 166, 0, 0);
INSERT INTO cvterm VALUES (167, 53, 'infraspecies', '', 167, 0, 0);
INSERT INTO cvterm VALUES (168, 31, 'Publication', '', 168, 0, 0);
INSERT INTO cvterm VALUES (169, 55, 'Collection', 'A collection holding references to a number of related resources.', 169, 0, 0);
INSERT INTO cvterm VALUES (170, 55, 'member', 'A member of the collection', 170, 0, 0);
INSERT INTO cvterm VALUES (171, 55, 'description', 'A description.', 171, 0, 0);
INSERT INTO cvterm VALUES (172, 55, 'totalItems', 'The total number of items referenced by a collection.', 172, 0, 0);
INSERT INTO cvterm VALUES (173, 55, 'title', 'A title, often used along with a description.', 173, 0, 0);
INSERT INTO cvterm VALUES (174, 55, 'PartialCollectionView', 'A PartialCollectionView describes a partial view of a Collection. Multiple PartialCollectionViews can be connected with the the next/previous properties to allow a client to retrieve all members of the collection.', 174, 0, 0);
INSERT INTO cvterm VALUES (175, 57, 'name', 'The name of the item.', 175, 0, 0);
INSERT INTO cvterm VALUES (176, 57, 'alternateName', 'An alias for the item.', 176, 0, 0);
INSERT INTO cvterm VALUES (177, 57, 'comment', 'Comments, typically from users.', 177, 0, 0);
INSERT INTO cvterm VALUES (178, 57, 'description', 'A description of the item.', 178, 0, 0);
INSERT INTO cvterm VALUES (179, 57, 'publication', 'A publication event associated with the item.', 179, 0, 0);
INSERT INTO cvterm VALUES (180, 57, 'url', 'URL of the item.', 180, 0, 0);
INSERT INTO cvterm VALUES (181, 57, 'additionalType', 'An additional type for the item, typically used for adding more specific types from external vocabularies in microdata syntax. This is a relationship between something and a class that the thing is in.', 181, 0, 0);
INSERT INTO cvterm VALUES (182, 57, 'ItemPage', 'A page devoted to a single item, such as a particular product or hotel.', 182, 0, 0);
INSERT INTO cvterm VALUES (183, 58, 'biological sample', 'A biological sample analysed by a particular technology.', 183, 0, 0);
INSERT INTO cvterm VALUES (184, 58, 'protocol', 'A protocol is a process which is a parameterizable description of a process.', 184, 0, 0);
INSERT INTO cvterm VALUES (185, 59, 'clause', 'A clause consists of a subject and a predicate.', 185, 0, 0);
INSERT INTO cvterm VALUES (186, 59, 'references', 'references is a relation between one entity and the entity that it makes reference to by name, but is not described by it.', 186, 0, 0);
INSERT INTO cvterm VALUES (187, 59, 'position', 'A measurement of a spatial location relative to a frame of reference or other objects.', 187, 0, 0);
INSERT INTO cvterm VALUES (188, 59, 'annotation', 'An annotation is a written explanatory or critical description, or other in-context information (e.g., pattern, motif, link), that has been associated with data or other types of information.', 188, 0, 0);
INSERT INTO cvterm VALUES (189, 59, 'negation', 'NOT is a logical operator in that has the value true if its operand is false.', 189, 0, 0);
INSERT INTO cvterm VALUES (190, 59, 'vocabulary', 'A vocabulary is a collection of terms.', 190, 0, 0);
INSERT INTO cvterm VALUES (191, 59, 'email address', 'an email address is an identifier to send mail to particular electronic mailbox.', 191, 0, 0);
INSERT INTO cvterm VALUES (192, 59, 'assay', 'An assay is an investigative (analytic) procedure in laboratory medicine, pharmacology, environmental biology, and molecular biology for qualitatively assessing or quantitatively measuring the presence or amount or the functional activity of a target entity (the analyte) which can be a drug or biochemical substance or a cell in an organism or organic sample.', 192, 0, 0);
INSERT INTO cvterm VALUES (193, 59, 'cell line', 'A cell line is a collection of genetically identifical cells.', 193, 0, 0);
INSERT INTO cvterm VALUES (194, 59, 'study', 'A study is a process that realizes the steps of a study design.', 194, 0, 0);
SELECT pg_catalog.setval('cvterm_cvterm_id_seq', 194, true);
INSERT INTO cv_root_mview VALUES ('Address', 2920, 21, 'tripal_contact');
INSERT INTO cv_root_mview VALUES ('Affiliation', 2916, 21, 'tripal_contact');
INSERT INTO cv_root_mview VALUES ('Contact Type', 2902, 21, 'tripal_contact');
INSERT INTO cv_root_mview VALUES ('Publication', 168, 31, 'tripal_pub');
INSERT INTO cv_root_mview VALUES ('sequence_attribute', 775, 52, 'sequence');
INSERT INTO cv_root_mview VALUES ('sequence_collection', 1358, 52, 'sequence');
INSERT INTO cv_root_mview VALUES ('sequence_feature', 158, 52, 'sequence');
INSERT INTO cv_root_mview VALUES ('sequence_variant', 162, 52, 'sequence');
INSERT INTO cv_root_mview VALUES ('taxonomic_rank', 2842, 53, 'taxonomic_rank');
INSERT INTO db2cv_mview VALUES (5, 'germplasm_ontology', 2, 'CO_010', 4);
INSERT INTO db2cv_mview VALUES (8, 'efo', 8, 'EFO', 4);
INSERT INTO db2cv_mview VALUES (9, 'ero', 9, 'ERO', 2);
INSERT INTO db2cv_mview VALUES (13, 'IAO', 13, 'IAO', 3);
INSERT INTO db2cv_mview VALUES (46, 'ncbitaxon', 20, 'NCBITaxon', 1);
INSERT INTO db2cv_mview VALUES (45, 'ncit', 19, 'NCIT', 20);
INSERT INTO db2cv_mview VALUES (10, 'OBCS', 10, 'OBCS', 1);
INSERT INTO db2cv_mview VALUES (11, 'obi', 11, 'OBI', 2);
INSERT INTO db2cv_mview VALUES (12, 'ogi', 12, 'OGI', 1);
INSERT INTO db2cv_mview VALUES (41, 'sbo', 15, 'SBO', 3);
INSERT INTO db2cv_mview VALUES (59, 'SIO', 33, 'SIO', 10);
INSERT INTO db2cv_mview VALUES (52, 'sequence', 24, 'SO', 2386);
INSERT INTO db2cv_mview VALUES (42, 'swo', 16, 'SWO', 1);
INSERT INTO db2cv_mview VALUES (53, 'taxonomic_rank', 25, 'TAXRANK', 61);
INSERT INTO db2cv_mview VALUES (21, 'tripal_contact', 26, 'TCONTACT', 28);
INSERT INTO db2cv_mview VALUES (31, 'tripal_pub', 27, 'TPUB', 212);
INSERT INTO db2cv_mview VALUES (44, 'uo', 18, 'UO', 1);
INSERT INTO db2cv_mview VALUES (7, 'EDAM', 4, 'data', 21);
INSERT INTO db2cv_mview VALUES (6, 'dc', 3, 'dc', 1);
INSERT INTO db2cv_mview VALUES (55, 'hydra', 29, 'hydra', 6);
INSERT INTO db2cv_mview VALUES (2, 'local', 14, 'local', 35);
INSERT INTO db2cv_mview VALUES (14, 'organism_property', 14, 'local', 11);
INSERT INTO db2cv_mview VALUES (15, 'analysis_property', 14, 'local', 1);
INSERT INTO db2cv_mview VALUES (16, 'tripal_phylogeny', 14, 'local', 4);
INSERT INTO db2cv_mview VALUES (23, 'featuremap_units', 14, 'local', 5);
INSERT INTO db2cv_mview VALUES (24, 'featurepos_property', 14, 'local', 2);
INSERT INTO db2cv_mview VALUES (25, 'featuremap_property', 14, 'local', 8);
INSERT INTO db2cv_mview VALUES (26, 'library_property', 14, 'local', 2);
INSERT INTO db2cv_mview VALUES (27, 'library_type', 14, 'local', 6);
INSERT INTO db2cv_mview VALUES (28, 'project_property', 14, 'local', 2);
INSERT INTO db2cv_mview VALUES (29, 'study_property', 14, 'local', 1);
INSERT INTO db2cv_mview VALUES (38, 'tripal_analysis', 14, 'local', 2);
INSERT INTO db2cv_mview VALUES (39, 'nd_experiment_types', 14, 'local', 2);
INSERT INTO db2cv_mview VALUES (40, 'nd_geolocation_property', 14, 'local', 1);
INSERT INTO db2cv_mview VALUES (1, 'null', 1, 'null', 1);
INSERT INTO db2cv_mview VALUES (4, 'chado_properties', 1, 'null', 1);
INSERT INTO db2cv_mview VALUES (7, 'EDAM', 6, 'operation', 5);
INSERT INTO db2cv_mview VALUES (47, 'rdfs', 21, 'rdfs', 3);
INSERT INTO db2cv_mview VALUES (57, 'schema', 31, 'schema', 8);
INSERT INTO db2cv_mview VALUES (58, 'sep', 32, 'sep', 2);
INSERT INTO db2cv_mview VALUES (60, 'synonym_type', 34, 'synonym_type', 4);
