diff --git a/ahah_helper.module b/ahah_helper.module
index dfd222a..efe3f55 100644
--- a/ahah_helper.module
+++ b/ahah_helper.module
@@ -132,6 +132,13 @@ function ahah_helper_generic_submit($form, &$form_state) {
  */
 function ahah_helper_real_submit($form, &$form_state) {
   unset($form_state['storage']);
+  
+  // Call FormAPI's standard submit handler for the form.
+  $form_id = $form['form_id']['#value'];
+  $function = $form_id . '_submit';
+  if (function_exists($function)) {
+    $function($form, $form_state);
+  } 
 }
 
 /**
